### PR TITLE
[Testing] Test cases for runtime entry function verification.

### LINKED ISSUE #252: [Testing] Test cases for runtime entry function verification.
We should write test cases for `runtime verification of the entry function`. 
These test cases can be divided into two categories: `successful verification` and `failed verification`. 
Additionally, multiple test scenarios should be constructed for each category.

Reference the Move test cases at: https://github.com/move-language/move/tree/main/language/move-bytecode-verifier/transactional-tests/tests.

For example:
- entry_function_with_return_value.move
- entry_function_invalid_struct.move
- entry_function_valid_struct.move
- entry_function_invalid_reference.move

In the Rooch project, we can write the test code in this directory: `crates/rooch-integration-test-runner/tests/cases`.

### PR DESCRIPTION
**Pull request by WGB5445**
Part of #252

### CODE REVIEW

File: crates/rooch-integration-test-runner/tests/cases/entry_function_invalid_reference.move [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    entry public fun test_entry_function_invalid_reference_signer( _: & signer ){
+        
+    }
+
+    entry public fun test_entry_function_invalid_reference_mut_signer( _: &mut signer ){
+        
+    }
+
+    entry public fun test_entry_function_invalid_reference_storage_context( _: & storage_context::StorageContext ){
```
Comment (steelgeek091) [inline crates/rooch-integration-test-runner/tests/cases/entry_function_invalid_reference.move:17]: Immutable references of the StorageContext type are allowed.

File: crates/rooch-integration-test-runner/tests/cases/entry_function_invalid_struct.move [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+//# publish
+module creator::test {
+    use std::string;
+    use std::ascii;
+    use moveos_std::storage_context;
+    use moveos_std::object_id;
+    use moveos_std::tx_context;
+
+
+    entry public fun test_entry_function_invalid_struct_string( _str: string::String ){
```
Comment (steelgeek091) [inline crates/rooch-integration-test-runner/tests/cases/entry_function_invalid_struct.move:12]: String type is allowed.

**Review by steelgeek091 — CHANGES_REQUESTED** [review]

**Review by jolestar — COMMENTED** [review]
Put the test cases into `rooch-framework-tests`, and we also need the negative test cases.

File: crates/rooch-framework-tests/tests/cases/entry_function/entry_function_with_return_value.move [hunk 3]
```diff
@@ -0,0 +1,54 @@
+//# init --addresses creator=0x42
+
+//# publish
+module creator::test {
+
+    entry public fun test_entry_function_return_value_u8():u8{
```
Comment (jolestar) [inline crates/rooch-framework-tests/tests/cases/entry_function/entry_function_with_return_value.move:6]: I think the entry function should not return values. @WGB5445  @steelgeek091

Comment (steelgeek091) [reply crates/rooch-framework-tests/tests/cases/entry_function/entry_function_with_return_value.move:6]: Yes, the verifier also checks inside if the entry function has a return value, an error will be thrown. So here, the error message should be determined in the exp file.

https://github.com/rooch-network/rooch/blob/925ca13c486a6b81c8f866d2762ff5b55727ab1b/moveos/moveos-verifier/src/verifier.rs#L153

Comment (WGB5445) [reply crates/rooch-framework-tests/tests/cases/entry_function/entry_function_with_return_value.move:6]: But there is no error reported in the actual exp file.
It should be a problem when publishing
I think this needs to be fixed in another pr

File: crates/rooch-integration-test-runner/tests/cases/entry_function_invalid_struct.move
[the remark this answers is not in the document]
Comment (WGB5445) [reply crates/rooch-integration-test-runner/tests/cases/entry_function_invalid_struct.move:1]: > put all test case in rooch-framework-tests, do not repeat in rooch-integration-test-runner.

 Okay, I have moved all the tests to room framework tests

**Review by jolestar — APPROVED** [review]
LGTM, these test cases need to be refactored after the verifier resolves the script verification issue.

**Review by steelgeek091 — APPROVED** [review]

**Pull request merged**